src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1404,7 +1404,8 @@ internal static bool RequiresValidScopedOverrideForRefSafety(MethodSymbol? metho
             //   - The method returns a `ref struct` or returns a `ref` or `ref readonly`, or the method has a `ref` or `out` parameter of `ref struct` type, and
             //   ...
             int nRefParametersRequired;
-            if (method.ReturnType.IsRefLikeOrAllowsRefLikeType() ||
+            if ((overrideThisParameter is { RefKind: RefKind.Ref or RefKind.Out } && overrideThisParameter.Type.IsRefLikeOrAllowsRefLikeType()) ||
```

Comment:
Should we also check scoped mismatch of the receiver in CheckValidScopedOverride?

Reply:
I do not think there is a need to do this. To my understanding, the only way to adjust the scoped-ness of a this parameter is to use `[UnscopedRef]`, and an unsafe difference between `[UnscopedRef]` on interface and implementation is already blocked. For example: https://github.com/dotnet/roslyn/blob/71cc05dad5f064ebe57814a727bed959737306f2/src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs#L941

I believe similar enforcement already exists for other forms of unsafe `[UnscopedRef]` differences.